### PR TITLE
feat(rules): Adding rules for the rpc_firewall 

### PR DESCRIPTION
**Pull request by sagiesec**
A set of rules, which take as their source event logs produced by the [rpc firewall](https://github.com/zeronetworks/rpcfirewall). 
These detect multiple lateral movement techniques, privilege escalation attacks and other RPC based attacks.

### CODE REVIEW

**Comment by Neo23x0** [comment]
It's a non-builtin software. We did something similar in the past with Sysmon, later transformed all Sysmon based rules into generic categories and therefore I am not sure if we should repeat that with RPF Firewall. 

Maybe we should integrate product specific rules in a separate folder. 
We have "cleaned up" the `/rules/windows` folder so that only contains generic rules or rules for built-in services. 

We, @thomaspatzke, @frack113 and I should be aligned on that question before we merge them into this folder.

**Comment by sagiesec** [comment]
Thanks for the quick response. Please let me know which location works

**Comment by thomaspatzke** [comment]
I would prefer a location outside of the main windows hierarchy for the reasons stated by Florian, e.g. `rules/third_party/windows/rpc_firewall`. Perhaps there's something better than third_party, but it should be clear that some additional log source is required here.

Can the rules be generalized, like it was done for process creation? Are there similar log sources?

By the way, nice ruleset and concept!

**Review by thomaspatzke — COMMENTED** [review]
As stated in the discussion.

**Comment by sagiesec** [comment]
I'm not sure what you mean by "generalized". 
The RPC Firewall generates these rules, and there aren't log sources which give the same information that I'm aware of

**Comment by frack113** [comment]
Hello,
there is allready an `application` folder why not use it ?
```yaml
    product: rpc_firewall
    category: application
```

**Comment by sagiesec** [comment]
Will do

**Comment by sagiesec** [comment]
Am I wrong, or is the test-sima failing on other rules...?

**Comment by frack113** [comment]
> Am I wrong, or is the test-sima failing on other rules...?

Your rules/linux/other/lnx_security_tools_disabling_syslog.yml is not the sane than the master one  #2576

For me it can not be `product: windows` but `product: rpc_firewall`.
Otherwise every rule for an application in a windows pc will get a "product: windows"

In the same logic, I have to move the EDR rule and antivirus rules ...

File: tests/test_rules.py [hunk 1]
```diff
@@ -625,6 +625,7 @@ def test_title(self):
                          "There are rules with non-conform 'title' fields. Please check: https://github.com/SigmaHQ/sigma/wiki/Rule-Creation-Guide#title")
 
     def test_invalid_logsource_attributes(self):
+        print("Testing invalid log sources...\n")
```
Comment (frack113) [inline tests/test_rules.py:628]: Please dont change code in a rule PR.
Thanks

File: tools/config/splunk-windows.yml [hunk 2]
```diff
@@ -112,5 +112,10 @@ logsources:
     service: smbclient-security
     conditions:
       source: 'WinEventLog:Microsoft-Windows-SmbClient/Security'
+  windows-rpc-firewall:
+        product: windows
+        service: rpc-firewall
+        conditions:
```
Comment (frack113) [inline tools/config/splunk-windows.yml:118]:
```yaml
windows-rpc-firewall:
   product: rpc_firewall
   category: application
```

**Review by Neo23x0 — APPROVED** [review]

**Pull request merged**